R/utilities-qualification.R

```diff
@@ -0,0 +1,235 @@
+      sep = .Platform$file.sep
+    )
+
+    sectionMarkdown <- sprintf("%0.3d_%s.md", sectionIndex, removeForbiddenLetters(section$Title))
+
+    # section data.frame with every useful information
+    sectionOut <- data.frame(
+      id = section$Id,
+      title = section$Title,
+      content = section$Content %||% NA,
+      index = sectionIndex,
+      path = sectionPath,
+      md = sectionMarkdown,
+      level = sectionLevel,
+      stringsAsFactors = FALSE
+    )
+    sectionsOut <- rbind.data.frame(sectionsOut, sectionOut, stringsAsFactors = FALSE)
+
+    # If subsections are included and not empty
+    # Update sectionsOut data.frame
+    if (!isOfLength(section$Sections, 0)) {
+      sectionsOut <- sectionsAsDataFrame(
+        sectionsIn = section$Sections,
+        sectionsOut = sectionsOut,
+        parentFolder = sectionPath,
+        sectionLevel = sectionLevel + 1
+      )
+    }
+  }
+  return(sectionsOut)
+}
+
+#' @title createSectionOutput
+#' @description Create the Sections output and their markdown content
+#' @param configurationPlan A `ConfigurationPlan` object
+#' @param logFolder path where logs are saved
+#' @return Names of created appendices
+createSectionOutput <- function(configurationPlan, logFolder = getwd()) {
+  # Intro
+  resetReport(configurationPlan$getIntroMarkdown())
+  appendices <- configurationPlan$getIntroMarkdown()
+  for (intro in configurationPlan$intro) {
+    configurationPlan$copyIntro(intro, logFolder = logFolder)
+  }
+  # Sections
+  for (sectionIndex in configurationPlan$sections$index) {
+    sectionId <- configurationPlan$sections$id[sectionIndex]
+    # Create section output path
+    dir.create(configurationPlan$getSectionPath(sectionId), showWarnings = FALSE, recursive = TRUE)
+    # Initialize markdown appendices
+    resetReport(configurationPlan$getSectionMarkdown(sectionId), logFolder = logFolder)
+    appendices <- c(appendices, configurationPlan$getSectionMarkdown(sectionId))
+    # Possibility to add title to section content if not defined in sections
+    # addTextChunk(configurationPlan$getSectionTitle(sectionId),
+    #             fileName = configurationPlan$getSectionMarkdown(sectionId),
+    #             logFolder = logFolder)
+
+    # Add info from content to section content
+    configurationPlan$copySectionContent(sectionId, logFolder = logFolder)
+  }
+  # Inputs
+  for (input in configurationPlan$inputs) {
+    configurationPlan$copyInput(input, logFolder = logFolder)
+  }
+  return(appendices)
+}
+
+
+
+#' @title getOutputsTimeProfile
+#' @description Get a vector of output paths that are to be used in a time profile plot
+#' @param plot is a descriptor of the plot that is read from the `ConfigurationPlan` that is generated by the configuration plan json
+#' @return A vector of output paths that are to be used in the time profile plot descriptor `plot`
+getOutputsTimeProfile <- function(plot) {

```

Comment:
I think the function name should be more explicit to clarify that "plot" is an object from the config plan,
something like: getOutputsFromTimeProfileConfiguration

Reply:
Changed